sdk/identity/azure-identity/azure/identity/_credentials/client_secret.py

```diff
@@ -24,9 +24,10 @@ class ClientSecretCredential(ClientCredentialBase):
     :keyword cache_persistence_options: Configuration for persistent token caching. If unspecified, the credential
         will cache tokens in memory.
     :paramtype cache_persistence_options: ~azure.identity.TokenCachePersistenceOptions
-    :keyword List[str] additionally_allowed_tenants: Optional additional tenant ids for which the credential
-        may acquire tokens. Add the wildcard value "*" to allow the credential to acquire tokens for
-        any tenant the application is installed.
+    :keyword List[str] additionally_allowed_tenants: Specifies tenants in addition to the specified for which the
+        credential may acquire tokens. Add the wildcard value "*" to allow the credential to acquire tokens for
+        any tenant the logged in account can access. If no value is specified for tenant_id" this option will have
+        no effect, and the credential will acquire tokens for any requested tenant.
```

Comment:
`tenant_id` is not an optional parameter for `ClientSecretCredential`
```suggestion
    :keyword List[str] additionally_allowed_tenants: Specifies tenants in addition to the specified "tenant_id" for which the
        credential may acquire tokens. Add the wildcard value "*" to allow the credential to acquire tokens for
        any tenant the logged in account can access.
```